### DISCUSSION TITLE
Custom Plugin with Docker

### DISCUSSION THREAD

**Discussion by waytoharish** [Kong/kong#7649] [Help] [answered]
Hi Team,
I am trying to write a custom plugin in go. I am able to add that plugin to new container but not to the running Container.
In RealWorld we could not recreate the new Container when we introduce the new plugin. Please could someone help me how to solve the same.

Everytime when i have to run docker build to build the new container with my Custom Plugin.

**Comment by kikito** [comment, marked as answer]
Hi @waytoharish, I am not sure that what you want to accomplish is possible.
Updating a custom Lua plugin already requires rebuilding the running docker Container. It is similar to changing a running App's source code. When this happens on a "RealWorld" app that often requires recreating the Container on which it is running. Go plugins are even more complicated than that since they require a compilation step.
